[CI] musl build fails

@japaric hey, sorry to annoy you again with this :confused: 

Recently my musl build started failing and I am not sure what is going on. The [travis log](https://travis-ci.org/azerupi/mdBook/jobs/148857608) says:

```
error: linking with `cc` failed: exit code: 1
```

And a little further

```
note: /usr/bin/ld: BFD (GNU Binutils for Ubuntu) 2.22 internal error, aborting at ../../bfd/reloc.c line 443 in bfd_get_reloc_size
/usr/bin/ld: Please report this bug.

collect2: ld returned 1 exit status

error: aborting due to previous error
error: Could not compile `mdbook`.
```

Do you have an idea on how I might solve this?

--- a/.travis.yml
+++ b/.travis.yml
@@ -57,15 +57,6 @@ matrix:
     - os: linux
       env: TARGET=x86_64-unknown-linux-musl CHANNEL=nightly
 
-  # Musl builds fail due to a bug in Rust (https://github.com/azerupi/mdBook/issues/158)
-  allow_failures:
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-musl CHANNEL=stable
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-musl CHANNEL=beta
-    - os: linux
-      env: TARGET=x86_64-unknown-linux-musl CHANNEL=nightly
-
 install:
   - export PATH="$PATH:$HOME/.cargo/bin"
   - bash ci/install.sh

